Rules.py

```diff
@@ -331,160 +325,336 @@ def global_rules(world):
 
     set_rule(world.get_location('Swamp Tingle Woodfall Map'), lambda state: state.can_pop_balloon())
     set_rule(world.get_location('Swamp Tingle Snowhead Map'), lambda state: state.can_pop_balloon())
-    # I can't actually remember the two maps tingle sells, I'll have to just go check that
+    set_rule(world.get_location('Swamp Tingle Pic'), lambda state: state.can_use('Picto Box'))
 
     # set_rule(world.get_location('Swamp Path Rupee Pit Chest'), lambda state: True)
     # there's apparently 20 rupees in a pit, it's probly just open, but I need to play again to check the details
 
     set_rule(world.get_location('Swamp Shooting Gallery Quiver Prize'), lambda state: state.can_use('Bow'))
-    set_rule(world.get_location('Swamp Shooting Gallery HP Prize'), lambda state: state.can_use('Bow') or state.options('NoHardestArchery'))
+    set_rule(world.get_location('Swamp Shooting Gallery HP Prize'), lambda state: state.can_use('Bow'))
+    # or state.options('NoHardestArchery')
+    # that doesn't work now that I think about it
+    # it means if the option is checked, this is treated as open, obviously wrong
+    # otherwise nothing can be placed there
+    # so todo: determine how to restrict progression items from this spot if the option is selected
+    # (there's probly a mechanism for this in testrunner's build)
+
+    set_rule(world.get_entrance('Swamp Path To Southern Swamp (Cleaned)'), lambda state: state.event('Beat Odolwa'))
 
     ## Southern Swamp
-    set_rule(world.get_location('Swamp Tourist Roof HP'), lambda state: state.can_reach(world.get_location('Swamp Business Scrub')) or state.can('SomeWeirdGoronTrickIDunno'))
+    set_rule(world.get_location('Swamp Tourist Roof HP'), lambda state: state.has('Town Title Deed') or state.can('Goron Boost'))
```

Comment:
If you hand over the Town Title Deed, you'd also need Deku form, right?

Reply:
human form for that one, but yes